### DISCUSSION TITLE
[2024-02-05] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#8974] [Weekly Demo Day]
In the last week, we've delivered several important fixes to GUI2, making it much more usable. We're almost done with catching-up to the old GUI, while bringing more useful features (like text editor, better component browser etc.) 

The last steps yet to be done are:
* Remove unnecessary evaluations when previewing CB
* Bringing back the code editor
* Fix comments starting with `##`.

Of course, new issues may arise. All are tracked in #8745

**Comment by MichaelMauderer** [comment]
* #8904 
* #8917
* Currently investigating regression of E2E edge tests.
* Then E2E test for Loading table visualization with data as default one for `Table` output type -

**Comment by vitvakatu** [comment]
Things presented on the last demo got merged:
- https://github.com/enso-org/enso/pull/8873

A bunch of small fixes for various bugs landed:
- https://github.com/enso-org/enso/pull/8887
- https://github.com/enso-org/enso/pull/8902
- https://github.com/enso-org/enso/pull/8933 

Some things in progress:
- https://github.com/enso-org/enso/pull/8950 is on review – we’ve added a small arrow under each dropdown (on the video) as a temporary fix of interactions between different widget kinds.
- Working on a proper fix for broken dropdown placement logic: https://github.com/enso-org/enso/issues/8932 

https://github.com/enso-org/enso/assets/6566674/d8b28c60-b210-486f-8c93-c6041a2062e3

**Comment by farmaazon** [comment]
* Helped @kazcw with delivering AST synchronizing, which solves several issues (most prominently: the flickering and setting values on set/filter arguments): I added a very temporary mechanism for packaging rust FFI wasm to electron client (as ydocs server now uses it).
* #8909 - it should help with preview evaluation overload, but still not sure if enough.
* #8947 - E2E tests should be much more reliable on CI (and locally too).
* #8972 waits for review - it's about the missing connections in Geospatial Analysis example.
* #8946
   * `Enter` key accepts the edit in Numeric and Text input
   * Newly added node are selected
   * Placement of new nodes takes opened visualizations into account.

[Screencast from 2024-02-02 13-02-26.webm](https://github.com/enso-org/enso/assets/3919101/0f328824-1cbc-4b07-988e-dbf17b94dc2f)

**Comment by kazcw** [comment]
Smart edits:
- Implemented automatic parentheses (fixes #8884) [merged in 8893].

Support for Code Input Widget (#8753) and other planned features:
- Moved metadata into AST Y.Js objects (#8237) [merged in 8893].

Y.Js tree-sync necessities [merged in 8893]:
- Added IdMap repair to Y.Doc server.
- Added a reparse fallback if edit tree fails validation.

Text edits (#8238):
- #8957, #8956 -- Simplify module-update data flow to avoid a problem with new AST/text-sync API.

**Comment by mwu-tow** [comment]
* CI checks on develop commits https://github.com/enso-org/enso/pull/8901
* aarch64 macOS release builds (https://github.com/enso-org/enso/pull/8864 & https://github.com/enso-org/enso/pull/8944)
  * Still the random code signing issues need further investigation.

There was high amount of transient failures in CI, both on the Engine and GUI side. Worst offenders should be addressed right now. Please keep reporting any new issues you encounter.

**Comment by hubertp** [comment]
Relevant engine changes:
- Don't cancel pending visualization's upserts https://github.com/enso-org/enso/pull/8853
